packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,246 @@
+      ? // remove the dragged item from its position if it is not coming from
+        // the hidden section (it is a swap)
+        [
+          ...selectedColumns.slice(0, dragResult.source.index),
+          ...selectedColumns.slice(dragResult.source.index + 1),
+        ]
+      : selectedColumns;
+
+    const columns = isSwap ? selectedColumns : availableColumns;
+    const draggedColumn = columns.find(
+      (col) => col.key === dragResult.draggableId
+    );
+
+    // push the column in the new position
+    onUpdateColumns([
+      ...items.slice(0, dragResult.destination.index),
+      draggedColumn,
+      ...items.slice(dragResult.destination.index),
+    ]);
+  }
+};
+
+export const ColumnManager = (props) => {
+  const { formatMessage } = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragStart = () => {
+    setIsDragging(true);
+  };
+
+  const hiddenColumns = differenceWith(
+    props.availableColumns,
+    props.selectedColumns,
+    (a, b) => a.key === b.key
+  );
+
+  return (
+    <CollapsibleMotion isDefaultClosed={false}>
+      {({ registerContentNode, containerStyles }) => (
+        <Container {...filterDataAttributes(props)}>
+          <Spacings.Stack scale="xs">
+            <HeaderContainer>
+              <Text.Headline as="h3" intlMessage={messages.title} />
+              <AccessibleButton
+                onClick={props.onClose}
+                label={formatMessage(messages.closeButtonLabel)}
+              >
+                <CloseIcon size="medium" />
+              </AccessibleButton>
+            </HeaderContainer>
+            <Spacings.Stack scale="s">
+              <Spacings.Inset scale="xs">
+                <div style={containerStyles}>
+                  <div ref={registerContentNode}>
+                    <DragDropContext
+                      onDragEnd={(dragResult) =>
+                        handleDragEnd(
+                          dragResult,
+                          props.onUpdateColumns,
+                          props.selectedColumns,
+                          props.availableColumns,
+                          setIsDragging
+                        )
+                      }
+                      onDragStart={handleDragStart}
+                    >
+                      <Spacings.Inline scale="m">
+                        <DroppableContainer isDragging={isDragging}>
+                          <Spacings.Stack>
+                            <Spacings.Inline scale="xs" alignItems="center">
+                              <EyeCrossedIcon size="medium" />
+                              <FieldLabel
+                                title={formatMessage(messages.hiddenColumns)}
+                              />
+                            </Spacings.Inline>
+                            {props.areHiddenColumnsSearchable && (
+                              <AsyncSelectInput
+                                id="hidden-columns-panel-search-input"
```

Comment:
This id is not used anywhere else. What is it needed for? Should we use an `AccessibleHidden` and a `Label` so we can link them for easier testing?

Reply:
Actually I wasn't using the id at all, it was a leftover from the original code in [packages-shared/core](https://github.com/commercetools/merchant-center-frontend/blob/0cdcc6a8d4fca3c5bc6538502fad42937f3fa274/packages-shared/core/components/column-manager/column-manager.js#L154)

Reply:
Try to avoid links to private repos please